hypothesis-python/tests/conjecture/test_data_tree.py

```diff
@@ -567,3 +568,10 @@ def buf(data):
         prefix = tree.generate_novel_prefix(Random())
         data = ConjectureData.for_buffer(prefix)
         assert data.draw_float(min_value, max_value, allow_nan=False) == expected_value
+
+
+def test_datatree_repr():
+    tree = DataTree()
+    observer = tree.new_observer()
+    observer.draw_boolean(True, was_forced=False, kwargs={"p": 0.5})
+    assert pretty.pretty(tree) == "boolean True {'p': 0.5}"
```

Comment:
Let's test some more complex trees too

Reply:
I added a more complicated tree. I can go all out testing this even more if you want, but I wasn't sure how useful testing an effectively debug-only feature would be.